alhimick-prog/hometask-4/my_hash.rb

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+# Trying to make class MyHash look like the original class Hash
+class MyHash
+  include Enumerable
+
+  KEY_POSITION = 0
+  VALEU_POSITION = 1
+  START_SIZE = 16
+  LOAD_FACTOR = 0.7
+
+  attr_reader :size
+
+  def initialize
+    @current_size = START_SIZE
+    @hash_array = Array.new(@current_size) { [] }
+    @size = 0
+    @old_array = nil
+    @position = nil
+  end
+
+  def [](key)
+    value = nil
+    @hash_array[define_position(key)].each do |item|
+      value = item[VALEU_POSITION] if item[KEY_POSITION].equal?(key)
+    end
+    value
+  end
+
+  def []=(key, value)
+    return if position_in_bucket_if_present_in_hash(key)
```

Comment:
@alhimick-prog why do we need to return? 
original hash: 
![image](https://user-images.githubusercontent.com/32943048/142881669-ce6f3818-fe09-4e74-8b43-b3f39bce612f.png)

your hash:
![image](https://user-images.githubusercontent.com/32943048/142881770-7f77ea36-2757-4c07-a21e-2a140ed520ae.png)

Reply:
@lincAlinc the conditions did not say to implement the overwrite function, so I set up protection against data overwriting using the existing key. It's not needed? Maybe it will increase the speed?